app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,109 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all
+  end
+
+
+  def show
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      head :not_found
+    end
+  end
+
+
+  def new
+    @task = Task.new
+  end
+
+
+  def create
+    @task = Task.new(task_params)
+
+    if @task.save
+      redirect_to root_path
+    else
+      render :new
+    end
+  end
+
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      head :not_found
+    end
+  end
+
+
+  def update
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      head :not_found
+    end
+
+    is_updated = @task.update(task_params)
+
+    if is_updated
+      redirect_to task_path(@task)
+    else
+      render :edit
+    end
+  end
+
+
+  def destroy
+    task = Task.find_by(id: params[:id])
+
+    if task.nil?
+      head :not_found
+    end
+
+    if task.destroy
+      redirect_to root_path
+    else
+      render :show
+    end
+  end
+
+
+  def complete
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
```

Comment:
Complete and incomplete do almost exactly the same thing. Could you combine them somehow - maybe make one action that toggles the completion status?